### ISSUE TITLE
unconvert, aligncheck and varcheck don't support vendoring

### ISSUE THREAD

**Issue by stevenh** [alecthomas/gometalinter#190]
If you try to use unconvert, aligncheck or varcheck they report bogus errors due to not correctly supporting vendored packages.

For example:
```
main.go:44:10:warning: undeclared name: errors (unconvert)
```

Where main.go clearly imports errors:
```go
import  "github.com/pkg/errors"
```

I understand this is a bug with those tools, not gometalinter, but until those tools are fixed I think they should be disabled by default or removed from gometalinter.

**Comment by alecthomas** [comment]
I think that's overkill. The majority of people don't use vendoring, but those who do can simply disable them. What might be useful is disabling them when `--vendor` is passed.

**Comment by alecthomas** [comment]
BTW, have you reported this to those tools?

**Comment by Dieterbe** [comment]
hey guys.  I haven't used gometalinter yet but here's a tip that might be useful.

I run go vet and other tools like so: `go vet $(go list ./... | grep -v /vendor/)` this is equivalent to `./...` but minus vendored directories.
this might solve / work around your problems too.

**Comment by GarethLewin** [comment]
@Dieterbe an alternative that we use is to name your vendor folder `_vendor` and then to symlink `vendor` to `_vendor`. go tools tend to not recurse into symlinked folders or folders starting with `_`

**Comment by stevenh** [comment]
I'm afraid that won't help as the issue is not it doesn't correctly check vendor'ed code its that it can't find vendor'ed packages hence the checks on the local code fail.

**Comment by stevenh** [comment]
I thought we'd already reported the issues upstream but clearly not, so thanks for the prod @alecthomas, this has been done now.
